test/dependency-manager/interpolation-validation.test.ts

```diff
@@ -0,0 +1,145 @@
+import { expect } from 'chai';
+import { buildSpecFromYml, ValidationErrors } from '../../src';
+import { registerInterpolation } from '../../src/dependency-manager/utils/interpolation';
+
+describe('interpolation-validation', () => {
+  const context = {
+    architect: {
+      tag: 'latest'
+    }
+  }
+
+  describe('validate build block', () => {
+    it('cannot use secret in build block', async () => {
+      const component_config = `
+        name: hello-world
+        secrets:
+          environment: prod
+        services:
+          api:
+            build:
+              args:
+                ENV: \${{ secrets.environment }}
+        `
+
+      const component_spec = buildSpecFromYml(component_config)
+
+      expect(() => {
+        registerInterpolation(component_spec, context)
+      }).to.be.throws(ValidationErrors);
+    });
+
+    it('cannot use conditional in build block', async () => {
+      const component_config = `
+        name: hello-world
+        services:
+          api:
+            build:
+              args:
+                \${{ if architect.environment == 'local' }}:
+                  ENV: prod
+        `
+
+      const component_spec = buildSpecFromYml(component_config)
+
+      expect(() => {
+        registerInterpolation(component_spec, context)
+      }).to.be.throws(ValidationErrors);
+    });
+
+    it('cannot use conditional around build block', async () => {
+      const component_config = `
+        name: hello-world
+        services:
+          api:
+            \${{ if architect.environment == 'local' }}:
+              build:
+                args:
+                  ENV: prod
+        `
+
+      const component_spec = buildSpecFromYml(component_config)
+
+      expect(() => {
+        registerInterpolation(component_spec, context)
+      }).to.be.throws(ValidationErrors);
+    });
+
+    it('cannot use conditional around service block with build block', async () => {
```

Comment:
If you have a local only service why could you not do this? This may be a  more advanced feature but I see no reason you could not do.
```
${{ if architect.build.args.something == 'local' }}:
```
Would also like to see some tests that include the args.

Reply:
You'd be able to use conditionals anywhere with `architect.build.tag` and `architect.build.args`. The breaking change would be preventing non-build time interpolation like environment. 

Here is an example of a valid conditional in the build block:
https://github.com/architect-team/architect-cli/pull/580/files#diff-ccf91dc9ee7a6fcf782de9f37462393ebf1f28f1a01e586ecb6e3bbd97e1034bR93